contracts/StakingContract.sol

```diff
@@ -23,65 +28,57 @@ contract StakingContract {
     uint256 interest
   );
 
-  constructor(Token token_) {
+  constructor(
+    Token token_,
+    uint256 annualPercentageYield_
+  )
+  {
     token = token_;
+    annualPercentageYield = annualPercentageYield_;
   }
 
   /**
-   * @notice Transfers an amount of ETH from the msg.sender to this contract if the sender has not 
-   * deposited before and the amount is greater than 0.01 ETH.
-   * @dev This function checks if the sender has already made a deposit, and the deposit value is
-   * greater than 0.01 ETH. If so, increases the sender's "etherBalanceOf" with the "value" being send,
-   * make the sender's status isDeposited "true" and records thh current block timestamp to a mapping.
-   * It also emits an event which has three (3) params, msg.sender, msg.value, and block.timestamp.  
+   * @notice Deposits funds into the Staking Contract
+   * @dev This function checks if the user has already made a deposit, and the deposit value 
+   * @param amount The amount of ETH being transferred
    */
-  function deposit() public payable {
-    // Checks if the user has made a deposit already
-    require(isDeposited[msg.sender] == false, "Deposit already active");
-    // Checks if the value being sent is greater than 0.01
-    require(msg.value >= 1e16, "Deposit must be greater than 0.01 ETH");
+  function deposit(uint256 amount) public payable {
+    require(amount > 0, "Cannot deposit 0");
+    token.transferFrom(msg.sender, address(this), amount);    
 
-    // Changes the mapping values for the user
-    etherBalanceOf[msg.sender] += msg.value;
-    isDeposited[msg.sender] = true;
-    depositStart[msg.sender] += block.timestamp;
+    UserInfo memory user = userInfo[msg.sender];
```

Comment:
This needs to be `storage`, in order for the changes you make below to be written in the contract's state